### ISSUE TITLE
Issue with GT data

### ISSUE THREAD

**Issue by robertopreste** [bihealth/vcfpy#130]
* vcfpy version: 0.12.1
* Python version: 3.7
* Operating System: Ubuntu 18.04.2, macOS 10.14.5

### Description

I integrated vcfpy into an application ([HmtNote](https://github.com/robertopreste/HmtNote)), and I use it to parse VCF files and add custom annotations. I encountered an issue regarding GT data when processing a VCF file formatted as: 

```
#CHROM  POS	ID	REF	ALT	QUAL	FILTER	INFO	        FORMAT	 TEST1
chrMT	 73	        .	A	G	.	        PASS	AC=2;AN=2	GT:HF	 0/1:0.994
```

### What I Did

When traversing the file using `vcfpy.Reader.from_path()` and manipulating each record, the following error shows up: 

```
 File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/hmtnote/classes.py", line 502, in annotate
    for record in self.reader:
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/reader.py", line 175, in __next__
    result = self.parser.parse_next_record()
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/parser.py", line 802, in parse_next_record
    return self.parse_line(self._read_next_line())
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/parser.py", line 793, in parse_line
    return self._record_parser.parse_line(line)
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/parser.py", line 467, in parse_line
    calls = self._handle_calls(alts, format_, arr[8], arr)
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/parser.py", line 479, in _handle_calls
    call = record.Call(sample, data)
  File "/home/ubuntu/test/ang_vcf/venv/lib/python3.7/site-packages/vcfpy/record.py", line 234, in __init__
    for allele in ALLELE_DELIM.split(self.data["GT"]):
TypeError: expected string or bytes-like object
```

I found a temporary solution and it works for in my specific case, but maybe the author might find something more suitable. The temporary fix involves replacing lines 232-234 in `record.py` as such: 

```
# original lines 232-234
if self.data.get("GT", None) is not None: 
    self.gt_alleles = []
    for allele in ALLELE_DELIM.split(self.data["GT"]):

# my replacement 
if self.data.get("GT", None) is not None and len(self.data.get("GT", None)) != 0:
    self.gt_alleles = []
    for allele in ALLELE_DELIM.split(self.data["GT"][0]):
```

**Comment by holtgrewe** [comment]
@robertopreste Thanks for the report! Your VCF example is not complete, apparently, I had to add the `##fileformat` header line. I could not reproduce your problem.

**Could you setup a minimal working example with a complete minimal VCF file, a minimal python script to trigger the problem (or a fork of vcfpy with a test case that triggers the problem)?**

Try the branch `130-gt-data`. Then install all dependencies in `requirements/*.txt`. Then:

```
# pytest tests/test_vcfs_from_the_wild.py
rootdir: /mnt/c/Users/mholtgre/Documents/Development/vcfpy, inifile: setup.cfg
plugins: cov-2.7.1, sugar-0.9.2
collecting ...
 tests/test_vcfs_from_the_wild.py ✓                                                                                                                                                                                          100% ██████████
============================================================================================================= warnings summary =============================================================================================================
/home/manuel/miniconda3/envs/vcfpy/lib/python3.7/site-packages/pysam/__init__.py:16
/home/manuel/miniconda3/envs/vcfpy/lib/python3.7/site-packages/pysam/__init__.py:16
  /home/manuel/miniconda3/envs/vcfpy/lib/python3.7/site-packages/pysam/__init__.py:16: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    import pysam.libcsamfile as libcsamfile

-- Docs: https://docs.pytest.org/en/latest/warnings.html

----------- coverage: platform linux, python 3.7.3-final-0 -----------
Coverage XML written to file coverage.xml


Results (0.47s):
       1 passed
```


PS: you appear to be using bumpversion. You might be interested in the excellent versioneer package.

**Comment by robertopreste** [comment]
Hello, thanks for your response. 
I just made a pull request with the complete VCF file that causes the error I mentioned. The `tests/test_vcfs_from_the_wild.py` test works fine, but if you add a simple loop to iterate over each record in the given VCF file, it breaks and returns the error I described. (It also does the same thing with the small snippet of VCF I provided above, but maybe with the complete file you might be able to investigate it better.)

**Comment by holtgrewe** [comment]
Hi @robertopreste. I'm sorry for taking so long to answer. The issue is the following header:

```
##FORMAT=<ID=GT,Number=.,Type=String,Description="Genotype">
```

The issue is the `Number=.` The VCF Standard proposes the following see (1.6.2 in https://samtools.github.io/hts-specs/VCFv4.3.pdf)

```
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
```

tl;dr -- your file is noncompliant in my opinion. How should we proceed here?

**Comment by robertopreste** [comment]
Hello @holtgrewe, thanks for your answer. 
I will forward this to the colleague who gave me this VCF file, so that he can hopefully fix the format. FYI, since my colleague was in a hurry for some analyses, I'm currently using a custom version of your vcfpy package which addresses this issue with an ugly try-except clause, but I'm going to delete it as soon as this issue is fixed. 

I guess we can close this issue now. Thanks again for your help.
